### PR TITLE
Ubuntu/bionic: release 23.1

### PR DESCRIPTION
**Pull request by aciba90**
# DO NOT SQUASH

## Steps to reproduce

```sh
git fetch upstream --tags
git checkout -B upstream/ubuntu/bionic ubuntu/bionic
new_upstream_snapshot.py -c 23.1 -b 2008230
# fix debian/changelog
```
## Checks

- [x] build-package
- [x] sbuild
- [x] no new lintian errors / warnings

### CODE REVIEW

**Review by blackboxsw — APPROVED** [review]
LGTM
- ✔no missing diffs in d/changelog and upstream/ubuntu/bionic-22.4.x:debian/changelog
- sbuild/build-package success no new lintian errors
- good version for release `(23.1-0ubuntu0~18.04.1) bionic`

**Comment by aciba90** [comment]
Updated to include quilt patch making route definitions compatible with netplan 0.99.

File: debian/changelog [hunk 1]
```diff
@@ -1,12 +1,31 @@
-cloud-init (22.4-0ubuntu0~18.04.2) UNRELEASED; urgency=medium
+cloud-init (23.1-0ubuntu0~18.04.2) bionic; urgency=medium
+
+  * d/patches/netplan99-cannot-use-default.patch:
```
Comment (blackboxsw) [inline debian/changelog:3]: +1 on this changeset and the quilt patch to disable use of 'to: default'  and bumping cloud-init version to 23.1-0ubuntu0~18.04.2

**Comment by blackboxsw** [comment]
One unrelated issue is quilt push -a ; tox -e flake8; fails with `tests/unittests/distros/test_netconfig.py:1074:17: F841 local variable 'netplan_readonly' is assigned to but never used`   I'll file separate PRs against the ubuntu/* series to resolve this flake8 issue. in unittests due to quilt pathces  It doesn't break package build because we don't call the flake8 target during package build for ubuntu.

**Comment by aciba90** [comment]
Tagged and pushed.

**Pull request merged**